lib/dynflow/execution_plan/hooks.rb

```diff
@@ -62,7 +62,9 @@ def dup
         def run(execution_plan, action, kind)
           on(kind).each do |hook|
             begin
-              action.send(hook, execution_plan)
+              Executors.run_user_code do
```

Comment:
I know it is better to be on the safe side, but is it really necessary to do the cleanup after running every single hook? I mean there could be more hooks bound to a single execution plan state change

Reply:
Updated